### PR TITLE
Replace FilePointer with universal pathlib

### PR DESCRIPTION
**Pull request by delucchi-cmu**
See https://github.com/astronomy-commons/hipscat/pull/336

## Solution Description

Use fsspec's project, [universal-pathlib](https://github.com/fsspec/universal_pathlib) as a path provider. In this way, everywhere we currently pass in a `FilePointer` we can use a `Path` instead, and when we need to create remote connections, use `UPath`.

### CODE REVIEW

**Comment by delucchi-cmu** [comment]
Documentation issue will require https://github.com/astronomy-commons/hipscat-import/pull/387/, but I'd like to send for review for general comments.

File: src/lsdb/catalog/dataset/healpix_dataset.py [hunk 1]
```diff
@@ -408,10 +409,9 @@ def plot_pixels(self, projection: str = "moll", **kwargs):
 
     def to_hipscat(
         self,
-        base_catalog_path: str,
+        base_catalog_path: UPath,
```
Comment (smcguire-cmu) [inline src/lsdb/catalog/dataset/healpix_dataset.py:412]: Can we have the user facing functions as `UPath | str`? Like this and `read_hipscat`

File: src/lsdb/loaders/hipscat/abstract_catalog_loader.py [hunk 2]
```diff
@@ -105,3 +93,12 @@ def _get_kwargs(self) -> dict:
         if self.config.dtype_backend is not None:
             kwargs["dtype_backend"] = self.config.dtype_backend
         return kwargs
+
+
+def read_pixel(
```
Comment (smcguire-cmu) [inline src/lsdb/loaders/hipscat/abstract_catalog_loader.py:98]: Can we add a specific columns kwarg to this? The way `from_map` with dask expressions works is that if the method has a columns param then it use it for query optimization, but if not will just always load all columns.

**Review by smcguire-cmu — COMMENTED** [review]
Soo nice to be getting rid of storage options from everywhere

File: src/lsdb/catalog/dataset/healpix_dataset.py (hunk 1 above)
Comment (delucchi-cmu) [reply src/lsdb/catalog/dataset/healpix_dataset.py]: Switched a bunch of places to use `str | Path | UPath`.

**Review by smcguire-cmu — APPROVED** [review]

**Pull request merged**